src/bio/bio_monitor.c

```diff
@@ -116,6 +138,37 @@ bio_get_dev_state(struct nvme_health_stats *state, struct bio_xs_context *xs)
 	return rc;
 }
 
+/*
+ * Call internal method to get the BIO blobstore device state from the device owner
+ * xstream.
+ */
+int
+bio_get_bs_state(int *bs_state, struct bio_xs_context *xs)
+{
+	struct bs_state_msg_arg		bsm = { 0 };
+	int				rc;
+
+	rc = ABT_eventual_create(0, &bsm.eventual);
+	if (rc != ABT_SUCCESS)
+		return dss_abterr2der(rc);
+
+	bsm.xs = xs;
+
+	spdk_thread_send_msg(owner_thread(xs->bxc_blobstore),
+			     bio_get_bs_state_internal, &bsm);
```

Comment:
It's just an atomic integer read over bb_state, we can read it directly instead of sending message to owner thread.

Reply:
good catch, will update